bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/base/conditions/mining/LocalConditions.java

```diff
@@ -102,7 +103,12 @@ private ServiceInvocationResult<Collection<ConditionPack>> noLicenses(LicensedPr
 	}
 
 	private Collection<Path> licenses(LicensedProduct product) throws LicensingException {
-		return new FileCollection(base(product), scope).get();
+		Supplier<Path> base = base(product);
+		if (Files.exists(base.get())) {
+			return new FileCollection(base, scope).get();
+		} else {
+			return Collections.emptyList();
```

Comment:
Again, empty set is legal, where incorrect configuration (wrong path) is not. 

Supplier must provide existing path, and `LocalConditions` must rely on it. 

`base(LicensedProduct` declaration can benefit from doc mentioning the fact though.

Reply:
> Supplier must provide existing path, and `LocalConditions` must rely on it.

I'm not sure if this is really intended? If I understand it correctly in one application multiple different specific local conditions can be specified so that a user can choose where to place a license file. For example in `<user-home>/.passage/<path-to-product-version>` or within the installation location.
I think it is more user-friend if it is sufficient to place the license file in one of the available locations and does not a have to place it in all.
If that is the case, it is perfectly legal that for example `UserHomeCondition.base()` method returns a path that does not exists (if the user put the license file into the installation location maybe not even `<user-home>/.passage` must exists).

And I don't know a better place where to check that. Please suggest one if you do.

Alternatively the supplier could return an `Optional<Path>` and sub-classes would have to return an empty Optional if the path does not exists. But I suspect that this requires more code since each sub-class where it is not an error if the path is absent has to check the existence in its base() implementation.

Reply:
Condition mining does only mining.
Condition mining cannot do so, when equipped with not-existing path to mine over. 

Equip condition mining with existing path and be sure you can all conditions from the location. 
There are several `MinedConditions` implementations, each configure existing base path.

If you implement custom version of `MinedConditions` on base of `LocalConditions`, just make sure it never gets invalid base path. 

If you are not ready to guarantee this, you can create your own implementation of pure `MinedConditions` interface and rely on your custom assumptions in it.

Reply:
> Condition mining does only mining. Condition mining cannot do so, when equipped with not-existing path to mine over.

IMHO it should also be valid and would make sense to mine over a non-existing path and consider it as empty.

> Equip condition mining with existing path and be sure you get all conditions from the location. There are several `MinedConditions` implementations, each configure existing base path.

I cannot confirm that. For example the `UserHomeResidentConditions` uses paths, that do not necessarily exist (at least I cannot find any code path that ensures the existence).

And for example if I launch [Passages License-Operator in version 2.4.0-M3](https://download.eclipse.org/passage/downloads/milestone/2.4.0-M3/operator/) for the very first time, i.e. without having a license file and without having the agreements yet accepted so that `<user-home>/.passage/org.eclipse.passage.loc.operator.product/2.4.0/` does not exist, I get the following errors: 

![image](https://user-images.githubusercontent.com/44067969/171125165-ba4646be-a5bb-45a2-bea3-0e01ebea84ce.png)

IMHO it is valid that for example `<user-home>/.passage/org.eclipse.passage.loc.operator.product/2.4.0/` does not exist at the very first start and I think it would lead to a bad user experience to require that a user has to create those folders manually in advance.
And in general the path searched by the `UserHomeResidentConditions` could remain empty if the license file is placed at an alternative location and the product has no agreements. Therefore I think it would be quite inconvenient to require its existence.

Reply:
Have created personal ticket for Passage Operator startup issue: #1088